nodes/src/nodes/tool_http_request/IInstance.py

```
@@ -17,30 +17,178 @@
+
+        url = args.get('url')
+        if not url or not isinstance(url, str):
+            raise ValueError('url is required and must be a non-empty string')
+        if self.IGlobal.url_patterns and not any(p.search(url) for p in self.IGlobal.url_patterns):
+            raise ValueError(f'URL "{url}" does not match any allowed URL pattern.')
+
+        auth = args.get('auth')
+        if isinstance(auth, dict):
+            auth_type = (auth.get('type') or 'none').strip().lower()
+            if auth_type not in valid_auth_types:
+                raise ValueError(f'auth.type must be one of {sorted(valid_auth_types)}; got {auth_type!r}')
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)